include/aos/iam/modules/certmodule.hpp

```diff
@@ -0,0 +1,420 @@
+     *
+     * @param x509Provider provider of x509 certificates, csr, keys.
+     * @param hsm a reference to hardware security module.
+     * @param storage a reference to certificate storage.
+     * @return Error.
+     */
+    Error Init(crypto::x509::ProviderItf& x509Provider, HSMItf& hsm, StorageItf& storage);
+
+    /**
+     * Returns IAM module certificate type.
+     *
+     * @return const String&.
+     */
+    const String& GetCertType() const;
+
+    /**
+     * Returns information about certificate with specified issuer and serial number.
+     *
+     * @param issuer certificate issuer.
+     * @param serial serial number.
+     * @param resCert result certificate.
+     * @return Error.
+     */
+    Error GetCertificate(const Array<uint8_t>& issuer, const Array<uint8_t>& serial, CertInfo& resCert);
+
+    /**
+     * Owns the module.
+     *
+     * @param password certficate password.
+     * @return Error.
+     */
+    Error SetOwner(const String& password);
+
+    /**
+     * Removes all module certificates.
+     *
+     * @return Error.
+     */
+    Error Clear();
+
+    /**
+     * Generates private key.
+     *
+     * @param password owner password.
+     * @return RetWithError<crypto::PrivateKey*> .
+     */
+    RetWithError<crypto::PrivateKey*> CreateKey(const String& password);
+
+    /**
+     * Creates certificate request.
+     *
+     * @param subject certificate subject.
+     * @param privKey private key.
+     * @param[out] pemCSR result csr in PEM.
+     * @return Error.
+     */
+    Error CreateCSR(const Array<uint8_t>& subject, const crypto::PrivateKey& privKey, Array<uint8_t>& pemCSR);
+
+    /**
+     * Applies certificate to a module.
+     *
+     * @param pemCert certificate chain in PEM format.
+     * @param[out] info result certificate information.
+     * @returns Error.
+     */
+    Error ApplyCert(const Array<uint8_t>& pemCert, CertInfo& info);
+
+    /**
+     * Creates a self signed certificate.
+     *
+     * @param password owner password.
+     * @returns Error.
+     */
+    Error CreateSelfSignedCert(const String& password);
+
+private:
+    static constexpr auto cCertChainSize = AOS_CONFIG_CERTHANDLER_CERTS_CHAIN_SIZE;
+    static constexpr auto cMaxSelfSignedCertSize = AOS_CONFIG_CERTHANDLER_PEM_SELFSIGNED_CERT_SIZE;
+    static constexpr auto cPasswordLen = AOS_CONFIG_CERTHANDLER_PASSWORD_LEN;
+    static constexpr auto cDNStringLen = AOS_CONFIG_CERTHANDLER_DN_STRING_LEN;
+    static constexpr auto cValidSelfSignedCertPeriod = time::Years(100);
+
+    static constexpr auto cOidExtensionExtendedKeyUsage = "2.5.29.37";
+    static constexpr auto cOidExtKeyUsageClientAuth = "1.3.6.1.5.5.7.3.1";
+    static constexpr auto cOidExtKeyUsageServerAuth = "1.3.6.1.5.5.7.3.2";
+
+    using StorageCerts = StaticArray<CertInfo, cCertsPerModule>;
+    using DNString = StaticString<cDNStringLen>;
+    using ModuleCertificates = StaticArray<CertInfo, cCertsPerModule>;
```

Comment:
Why do we need different using for the same types? Name it to cover the both cases: `ModuleCerts` for example.